lib/publiccloud/instance.pm

```diff
@@ -395,13 +399,29 @@ sub softreboot
     die("Waiting for system down failed!") unless ($shutdown_time < $args{timeout});
     my $bootup_time = $self->wait_for_ssh(timeout => $args{timeout} - $shutdown_time, username => $args{username});
 
-    # Re-establish tunnel and switch back to previous console if TUNNELED
-    if ($tunneled) {
+    # Re-establish tunnel and switch back to previous console if needed
+    if (get_var("TUNNEL_AUTO_SSH")) {
         ssh_interactive_tunnel($self);
         die("expect ssh serial device to be active") unless (get_var('SERIALDEV') =~ /ssh/);
+        record_info("console", "$prev_console");
+        select_console($prev_console) if ($prev_console !~ /tunnel/);
+    } elsif ($tunneled) {
+        ssh_interactive_tunnel($self);
+        die("expect ssh serial device to be active") unless (get_var('SERIALDEV') =~ /ssh/);
+        # Connect to sut
+        select_console('root-console');
+        script_run('ssh -E /var/tmp/ssh_sut.log -vt sut', timeout => 0);
+        select_console('user-console');
+        script_run('ssh -E /var/tmp/ssh_sut.log -vt sut', timeout => 0);
+        sleep(5);
+        record_info("console", "$prev_console");
         select_console($prev_console) if ($prev_console !~ /tunnel/);
     }
 
+    # Tunnel smoke test: print hostname to check on which host we are
```

Comment:
if it is a test maybe we need to check if it is equal to something or NOT equal to something and fail otherwise ?